dataplane/dataplane.cpp

```diff
@@ -1242,51 +1335,56 @@ void cDataPlane::init_worker_base()
 	neighbor.update_worker_base(base_nexts);
 }
 
-void cDataPlane::SWRateLimiterTimeTracker()
+void cDataPlane::timestamp_thread()
 {
+	uint32_t prev_time = 0;
+
 	for (;;)
 	{
-		// seem to be sufficiently fast function for slowWorker whose threshold is 200'000 packets per second
-		std::chrono::high_resolution_clock::time_point curTimePointForSWRateLimiter = std::chrono::high_resolution_clock::now();
+		current_time = time(nullptr);
 
-		// is it time to reset icmpPacketsToSW counters?
-		if (std::chrono::duration_cast<std::chrono::milliseconds>(curTimePointForSWRateLimiter - prevTimePointForSWRateLimiter) >= std::chrono::milliseconds(1000 / config.rateLimitDivisor))
+		if (current_time != prev_time)
 		{
-			// the only place thread-shared variable icmpPacketsToSW is changed
-			for (cWorker* worker : workers_vector)
+			for (const auto& [socket_id, globalbase_atomic] : globalBaseAtomics)
 			{
-				__atomic_store_n(&worker->packetsToSWNPRemainder, config.SWNormalPriorityRateLimitPerWorker, __ATOMIC_RELAXED);
+				(void)socket_id;
+				globalbase_atomic->currentTime = current_time;
 			}
 
-			controlPlane->icmpOutRemainder = config.SWICMPOutRateLimit / config.rateLimitDivisor;
-
-			prevTimePointForSWRateLimiter = curTimePointForSWRateLimiter;
+			prev_time = current_time;
 		}
-		using namespace std::chrono_literals;
-		std::this_thread::sleep_for(100ms / config.rateLimitDivisor);
+
+		std::this_thread::sleep_for(std::chrono::milliseconds(500));
 	}
 }
 
-void cDataPlane::timestamp_thread()
+void cDataPlane::SWRateLimiterTimeTracker()
 {
-	uint32_t prev_time = 0;
-
 	for (;;)
 	{
-		current_time = time(nullptr);
+		using namespace std::chrono_literals;
+		// seem to be sufficiently fast function for slowWorker whose threshold is 200'000 packets per second
+		std::chrono::high_resolution_clock::time_point curTimePointForSWRateLimiter = std::chrono::high_resolution_clock::now();
 
-		if (current_time != prev_time)
+		// is it time to reset icmpPacketsToSW counters?
+		if (std::chrono::duration_cast<std::chrono::milliseconds>(
+		            curTimePointForSWRateLimiter - prevTimePointForSWRateLimiter) >= 1000ms / config.rateLimitDivisor)
 		{
-			for (const auto& [socket_id, globalbase_atomic] : globalBaseAtomics)
+			// the only place thread-shared variable icmpPacketsToSW is changed
+			for (cWorker* worker : workers_vector)
 			{
-				(void)socket_id;
-				globalbase_atomic->currentTime = current_time;
+
+				__atomic_store_n(&worker->packetsToSWNPRemainder, config.SWNormalPriorityRateLimitPerWorker, __ATOMIC_RELAXED);
 			}
```

Comment:
Ugh. Let's use `std::atomic` with `std::memory_order_relaxed`.

Change declaration of packetsToSWNPRemainder to:
```cpp
// will decrease with each new packet sent to slow worker, replenishes each N mseconds
 std::atomic<int32_t> packetsToSWNPRemainder; 
```

And use here
```cpp
worker->packetsToSWNPRemainder.store(config.SWNormalPriorityRateLimitPerWorker, std::memory_order_relaxed);
```

Reply:
Note that we also will need to change the following function:
```cpp
inline void cWorker::slowWorker_entry_normalPriority(rte_mbuf* mbuf,
                                                     const common::globalBase::eFlowType& flowType)
{
	/// @todo: worker::tStack

	if (basePermanently.SWNormalPriorityRateLimitPerWorker != 0 && __atomic_fetch_sub(&packetsToSWNPRemainder, 1, __ATOMIC_RELAXED) <= 0)
	{
		rte_pktmbuf_free(mbuf);
		counters[(uint32_t)common::globalBase::static_counter_type::slow_worker_normal_priority_rate_limit_exceeded]++;

		return;
	}

	dataplane::metadata* metadata = YADECAP_METADATA(mbuf);
	metadata->flow.type = flowType;

	if (rte_ring_sp_enqueue(ring_normalPriority, (void*)mbuf))
	{
		stats.ring_normalPriority_drops++;
		rte_pktmbuf_free(mbuf);
	}
	else
	{
		stats.ring_normalPriority_packets++;
	}
}
```
to
```cpp
inline void slowWorker_entry_normalPriority(rte_mbuf* mbuf, const common::globalBase::eFlowType& flowType)
{
        /// @todo: worker::tStack

        if (basePermanently.SWNormalPriorityRateLimitPerWorker != 0 && packetsToSWNPRemainder.fetch_sub(1, std::memory_order_relaxed) <= 0)
        {
            rte_pktmbuf_free(mbuf);
            counters[static_cast<uint32_t>(common::globalBase::static_counter_type::slow_worker_normal_priority_rate_limit_exceeded)]++;
            return;
        }

        auto* metadata = YADECAP_METADATA(mbuf);
        metadata->flow.type = flowType;

        if (rte_ring_sp_enqueue(ring_normalPriority, static_cast<void*>(mbuf)) != 0)
        {
            stats.ring_normalPriority_drops++;
            rte_pktmbuf_free(mbuf);
        }
        else
        {
            stats.ring_normalPriority_packets++;
        }
    }
```

Reply:
This can of worms is better handled in separate refactoring PR